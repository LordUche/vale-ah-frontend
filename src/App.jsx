import React from 'react';
import { hot } from 'react-hot-loader/root';

/**
 * Renders the App
 * @returns {JSX} App Component
 */
const App = () => <h1>Welcome to Naija Chop Chop</h1>;

export default hot(App);
