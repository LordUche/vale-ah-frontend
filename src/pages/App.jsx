import { hot } from 'react-hot-loader/root';
import React from 'react';

/**
 * Render a dummy page to test react app
 * @returns {JSX} Page Component
 */
const App = () => <h1>Welcome to Naija Chop Chop</h1>;

export default hot(App);
