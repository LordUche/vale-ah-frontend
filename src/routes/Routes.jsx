import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';

/**
 * @description - contains all the routes for the application
 * @param {object} props
 * @returns {route} - route
 */
const route = () => (
  <Fragment>
    <Switch>
      <Route path="/page1" render={() => <h3>Hello from page1</h3>} exact />
      <Route path="/page2" render={() => <h3>Hello from page2</h3>} exact />
    </Switch>
  </Fragment>
);

export default route;
