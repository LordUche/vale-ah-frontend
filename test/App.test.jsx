import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../src/App';
import Routes from '../src/routes/Routes';
import { mockStore } from './setupTests';

const wrapper = mount(
  <Provider store={mockStore({ auth: { isLoggedIn: false } })}>
    <App />
  </Provider>
);
describe('<App/> rendering', () => {
  it('should render one <Routes> component', () => {
    expect(wrapper.find(Routes)).toHaveLength(1);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
