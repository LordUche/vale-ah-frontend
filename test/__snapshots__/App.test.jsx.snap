// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<h1>
  Welcome to Naija Chop Chop
</h1>
`;
